test/test-emfile.c

```diff
@@ -75,6 +75,14 @@ TEST_IMPL(emfile) {
   ASSERT_EQ(errno, EMFILE);
   close(maxfd);
 
+#if defined(__ANDROID__)
+  /* Android connect syscall requires an extra file descriptor
```

Comment:
Can you elaborate? I'm intrigued :-)

Reply:
For sure. I don't think I will have the right terms - I even tried to find more details about it on the internet but I couldn't.

If I understood correctly, this test depletes file descriptors and close on to be used by `uv_tcp_connect`. 

Well, on Android, even though the fd 31 was available it was returning an EMFILE error. After some debugging I realized that [here](https://github.com/libuv/libuv/blob/f00d4b677556a007a28fcb0bb3548ecdb54a29ae/src/unix/tcp.c#L310), `connect` returns `-1` and sets the `errno` even though the `uv__stream_fd` returns the right file descriptor.

Not sure if it's an Android protection or something.

I then closed one more fd and it worked as expected.